### DISCUSSION TITLE
Query for a single tag

### DISCUSSION THREAD

**Discussion by aalmiray** [hub4j/github-api#1499] [Q&A] [answered]
Hello everyone, is it possible to query for the existence of a single tag, with or without an associated release?

So far I've seen methods to list all tags (which may be expensive in repos with too many tags, also don't need all the data per tag), and a method to find a release by tag (there might not be a release yet associated with said tag).

At the moment I'm using the option to list all tags then iterate til I find the tag or not.

**Comment by bitwiseman** [comment, marked as answer]
@aalmiray 
You can query for the existence of a `ref`. 

https://github.com/hub4j/github-api/blob/cf27f0cfc63ca42eb00a9c2e59e656cc099a4546/src/main/java/org/kohsuke/github/GHRepository.java#L1884

`repository.getRef("tags/" + tagName)`.  This removes the need to iterate.
